### PR TITLE
Add GCP instructions, minor AWS changes

### PR DESCRIPTION
**Pull request by devinmarieb**
# Description
New section to OIDC page for GCP.

# Reasons
We've added support for OIDC with GCP set up.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [x] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 1]
```diff
@@ -33,15 +33,13 @@ workflows:
...
+The https://openid.net/specs/openid-connect-core-1_0.html#Terminology[OpenID Provider] is unique to your organization. The URL is `\https://oidc.circleci.com/org/ORGANIZATION_ID`, where `ORGANIZATION_ID`` is the organization ID (a universally unique identifier) that represents your organization. You can find your CircleCI organization ID by navigating to **Organization Settings > Overview** on the https://app.circleci.com/[CircleCI web app].
 
-The OpenID Connect ID tokens issued by CircleCI have a fixed audience (see `aud` in the table below), which is also the Organization ID.
+The OpenID Connect ID tokens issued by CircleCI have a fixed audience (see `aud` in the table below), which is also the organization ID.
 
 == Format of the OpenID Connect ID token
 
-The OpenID Connect ID token contains the following standard https://openid.net/specs/openid-connect-core-1_0.html#IDToken[claims]:
+Th e OpenID Connect ID token contains the following standard https://openid.net/specs/openid-connect-core-1_0.html#IDToken[claims]:
```
Comment (thomasr8) [inline jekyll/_cci2/openid-connect-tokens.adoc:42]:
```suggestion
The OpenID Connect ID token contains the following standard https://openid.net/specs/openid-connect-core-1_0.html#IDToken[claims]:
```

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 2]
```diff
@@ -51,13 +49,13 @@ The OpenID Connect ID token contains the following standard https://openid.net/s
 | Description
 
 | `iss`
-| The issuer. The issuer is specific to the CircleCI organization in which the job is being run. Its value is `"https://oidc.circleci.com/org/<organization-id>"`, a    string, where `<organization-id>` is a UUID identifying the current job's project's organization.
+| The issuer. The issuer is specific to the CircleCI organization in which the job is being run. Its value is `"https://oidc.circleci.com/org/ORGANIZATION_ID"`, a    string, where `ORGANIZATION_ID` is a UUID identifying the current job's project's organization.
```
Comment (thomasr8) [inline jekyll/_cci2/openid-connect-tokens.adoc:52]:
```suggestion
| The issuer. The issuer is specific to the CircleCI organization in which the job is being run. Its value is `"https://oidc.circleci.com/org/ORGANIZATION_ID"`, a  string, where `ORGANIZATION_ID` is a UUID identifying the current job's project's organization.
```

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 3]
```diff
@@ -82,29 +80,32 @@ The OpenID Connect ID token also contains some https://openid.net/specs/openid-c
...
+
+The following AWS instructions are for:
 
 * A one-time configuration of your AWS account to trust CircleCI's OIDC tokens
 * Running a job that uses the OIDC token to interact with AWS
 
-=== Setting up AWS
+==== Setting up AWS
 
 You will need to allow your AWS account to trust CircleCI's OpenID Connect tokens. To do this, create an Identity and Access Management(IAM) identity provider, and an IAM role in AWS (this is a one time configuration).
```
Comment (thomasr8) [inline jekyll/_cci2/openid-connect-tokens.adoc:96]:
```suggestion
You will need to allow your AWS account to trust CircleCI's OpenID Connect tokens. To do this, create an Identity and Access Management(IAM) identity provider, and an IAM role in AWS (this is a one-time configuration).
```

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 4]
```diff
@@ -141,20 +142,146 @@ jobs:
...
 This uses https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_String[StringLike] to match the sub claim of CircleCI's OIDC token in your chosen project. Now, jobs in your other projects cannot assume this role.
+
+=== Google Cloud Provider
+
+The following GCP instructions are for:
+
+* A one-time configuration of your GCP settings to trust CircleCI's OIDC tokens
+* Running a job that uses the OIDC token to interact with GCP
+
+The Google Cloud CLI reads your configuration file which contains necessary information instructing Google Cloud to authenticate. You can read about external identity providers on https://cloud.google.com/iam/docs/configuring-workload-identity-federation#oidc[Google Cloud's docs].
```
Comment (thomasr8) [inline jekyll/_cci2/openid-connect-tokens.adoc:170]:
```suggestion
The Google Cloud CLI reads your configuration file, which contains necessary information instructing Google Cloud to authenticate. You can read about external identity providers on https://cloud.google.com/iam/docs/configuring-workload-identity-federation#oidc[Google Cloud's docs].
```

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 5]
```diff
@@ -141,20 +142,146 @@ jobs:
...
+
+==== Setting up GCP
+
+The GCP configuration file can be set up using the GCP UI. In the **Workload Identity Federation UI**, navigate to **Grant Access**, which will prompt the configuration, which can then be downloaded. You will need to create a file named `CIRCLE_OIDC_TOKEN_FILE`, which Google Cloud will read your identity token from.
+
+You will need your CircleCI organization ID, which can be found by navigating to **Organization Settings > Overview** on the https://app.circleci.com/[CircleCI web app].
+
+After navigating to the **Grant Access** section of the GCP UI, follow these steps to add CircleCI as an external identity provider:
+
+1. Navigate to the **IAM & Admin panel**
```
Comment (thomasr8) [inline jekyll/_cci2/openid-connect-tokens.adoc:180]: Looks like the numbering did not format correctly?

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 6]
```diff
@@ -141,20 +142,146 @@ jobs:
...
+| assertion['oidc.circleci.com/project-id']
+|===
++
+1. Navigate to **Service Account** in the IAM & Admin Panel to create a service account, and give appropriate permission
+1. Navigate back to **Workload Identity Federation** and select the provider from the table
+1. Click the **Grant access** button
+1. A modal will open and you will select the service account you created from the dropdown. This is the account that the token will impersonate, which grants all the associated permissions
+1. Under **Select principals**, you can add conditions, or leave the default
+1. Click **Save**. A pop-up will appear to ask you configure and **download** the configuration file. This file can also be downloaded later by navigating to **Connected Service Accounts**
+
```
Comment (thomasr8) [inline jekyll/_cci2/openid-connect-tokens.adoc:208]: numbering looks off here too?

**Review by thomasr8 — COMMENTED** [review]
@devinmarieb Looks good! Just a few minor comments and suggestions. Also, is there a reason why we are using both "yaml" and "yml"? Just curious about the difference.

File: jekyll/_cci2/openid-connect-tokens.adoc (hunk 6 above)
Comment (devinmarieb) [reply jekyll/_cci2/openid-connect-tokens.adoc:208]: @thomasr8 This is how auto-numbering works in asciidoc. If you label everything as 1, the numbers render in order. This makes it easy to update or change instructions without worrying about having to re-order or update numbers in the future.

**Comment by devinmarieb** [comment]
> @devinmarieb Looks good! Just a few minor comments and suggestions. Also, is there a reason why we are using both "yaml" and "yml"? Just curious about the difference.
There is no difference. We mostly use yaml in the docs, but sometimes when we get sent code from developers they use yml. Three letters is a holdover from when extensions were limited to only three letters. But it makes no actual difference at all.

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 7]
```diff
@@ -141,20 +142,146 @@ jobs:
...
+The following GCP instructions are for:
+
+* A one-time configuration of your GCP settings to trust CircleCI's OIDC tokens
+* Running a job that uses the OIDC token to interact with GCP
+
+The Google Cloud CLI reads your configuration file, which contains necessary information instructing Google Cloud to authenticate. You can read about external identity providers on https://cloud.google.com/iam/docs/configuring-workload-identity-federation#oidc[Google Cloud's docs].
+
+==== Setting up GCP
+
+The GCP configuration file can be set up using the GCP UI. In the **Workload Identity Federation UI**, navigate to **Grant Access**, which will prompt the configuration, which can then be downloaded. You will need to create a file named `CIRCLE_OIDC_TOKEN_FILE`, which Google Cloud will read your identity token from.
```
Comment (BentoDevBox) [inline jekyll/_cci2/openid-connect-tokens.adoc:174]: nit: the file name can be anything as long as it matches with what's in the `credential_source.file`
```
"credential_source": {
    "file": "CIRCLE_OIDC_TOKEN_FILE",
    "format": {
      "type": "text"
    }
  }
```

**Review by BentoDevBox — APPROVED** [review]
LGTM

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 8]
```diff
@@ -149,20 +154,228 @@ jobs:
...
+workflows:
+  main:
+    jobs: 
+      - gcp-oidc-defaults:
+          name: Generate Creds File and Authenticate
+          context: 
+          - gcp-oidc-dev
+```
+
+You can use multiple service accounts from the same GCP project as well as using multiple service accounts from multiple GCP projects. You can read about these methods and find an example on CircleCI's link:https://github.com/jtreutel/circleci-gcp-oidc-test#usage[example repository].
```
Comment (kbatuigas) [inline jekyll/_cci2/openid-connect-tokens.adoc:381]:
```suggestion
You can use multiple service accounts from the same GCP project as well as use multiple service accounts from multiple GCP projects. You can read about these methods and find an example on CircleCI's link:https://github.com/jtreutel/circleci-gcp-oidc-test#usage[example repository].
```
Maybe also "same GCP project, **or** use multiple service accounts..."?

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 9]
```diff
@@ -35,11 +35,9 @@ workflows:
 
 Consult your cloud service's documentation for how to add an identity provider. For example, AWS's https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html[Creating OpenID Connect (OIDC) identity providers], or Google Cloud Platform's https://cloud.google.com/iam/docs/configuring-workload-identity-federation#oidc[Configuring workload identity federation].
 
-The https://openid.net/specs/openid-connect-core-1_0.html#Terminology[OpenID Provider] is unique to your organization. The URL is `\https://oidc.circleci.com/org/<organization-id>`, where `<organization-id>` is the Organization ID (universally unique identifier) that represents your organization.
+The https://openid.net/specs/openid-connect-core-1_0.html#Terminology[OpenID Provider] is unique to your organization. The URL is `\https://oidc.circleci.com/org/ORGANIZATION_ID`, where `ORGANIZATION_ID`` is the organization ID (a universally unique identifier) that represents your organization. You can find your CircleCI organization ID by navigating to **Organization Settings > Overview** on the https://app.circleci.com/[CircleCI web app].
```
Comment (kbatuigas) [inline jekyll/_cci2/openid-connect-tokens.adoc:38]:
```suggestion
The https://openid.net/specs/openid-connect-core-1_0.html#Terminology[OpenID Provider] is unique to your organization. The URL is `\https://oidc.circleci.com/org/ORGANIZATION_ID`, where `ORGANIZATION_ID` is the organization ID (a universally unique identifier) that represents your organization. You can find your CircleCI organization ID by navigating to **Organization Settings > Overview** on the https://app.circleci.com/[CircleCI web app].
```
Doesn't really affect how it's rendered but just an extra backtick

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 10]
```diff
@@ -85,33 +83,38 @@ The OpenID Connect ID token also contains some https://openid.net/specs/openid-c
...
 
 * A one-time configuration of your AWS account to trust CircleCI's OIDC tokens
 * Running a job that uses the OIDC token to interact with AWS
 
 [#setting-up-aws]
-=== Setting up AWS
+==== Setting up AWS
 
-You will need to allow your AWS account to trust CircleCI's OpenID Connect tokens. To do this, create an Identity and Access Management(IAM) identity provider, and an IAM role in AWS (this is a one time configuration).
+You will need to allow your AWS account to trust CircleCI's OpenID Connect tokens. To do this, create an Identity and Access Management(IAM) identity provider, and an IAM role in AWS (this is a one-time configuration).
```
Comment (kbatuigas) [inline jekyll/_cci2/openid-connect-tokens.adoc:102]:
```suggestion
You will need to allow your AWS account to trust CircleCI's OpenID Connect tokens. To do this, create an Identity and Access Management (IAM) identity provider, and an IAM role in AWS (this is a one-time configuration).
```

**Review by kbatuigas — CHANGES_REQUESTED** [review]

**Review by kbatuigas — APPROVED** [review]
Looks great! 🎉

**Pull request merged**